compiler/test/fail_compilation/ice23781.d

```diff
@@ -0,0 +1,10 @@
+/**
+TEST_OUTPUT:
+---
+fail_compilation/ice23781.d(10): Error: variable `b` cannot be read at compile time
```

Comment:
`E1` is `immutable`, so isn't it expected behavior that the test cases compiles?

Reply:
Well, yes, it should, but it never worked. Will look more into it, but wanted to fix the obvious regression introduced in v2.068